lib/epochs.js

```diff
@@ -509,24 +511,46 @@ function epochNodeStream(ssb, groupId, opts = {}) {
 
   return deferredSource
 }
+function getRepeat(ssb, key, cb) {
+  ssb.db.get(key, (err, rootVal) => {
+    if (err) {
+      setTimeout(() => {
+        return getRepeat(ssb, key, cb)
+      }, 100)
+    } else {
+      return cb(null, rootVal)
+    }
+  })
+}
```

Comment:
:fire:  this needs a limit ... like 1 minute, or 100 tries?
100ms might want to be stretched out as well if not in tests?

Reply:
this is an eventually consistent situation, so might take 1ms or 3hrs. how about exponential backoff?

Reply:
switched to exponential backoff now, with a limit, i think it should give up after a few weeks or something with the current config lol

Reply:
The exponential backoff seems like an overkill to solve this race condition (box2 versus db2), and with hours or weeks between "cause" and "effect", this could become a debugging nightmare in the future.

What if you do only a minor exponential backoff (say, with a limit of 1 second, that should be enough to beat the basic db2 race conditions)? And then, what if you only trigger the "add missing members to preferred epoch" logic only if you know you have the group root?

Reply:
Mm yeah it is a bit weird the long delay. I think I'll just change the backoff to something short even if that could in theory introduce some other problem in the future, but that's for other new tests/reqs/bug reports to figure out.

Reply:
done

Reply:
didn't test this in the demo which i think is where it needs testing, let's see if it needs adjusting though